docs/entities/transactions/transaction-status.schema.json

```diff
@@ -3,5 +3,5 @@
   "title": "TransactionStatus",
   "description": "Status of the transaction",
   "type": "string",
-  "enum": ["success", "pending", "abort_by_response", "abort_by_post_condition"]
+  "enum": ["success", "pending", "abort_by_response", "abort_by_post_condition", ""]
```

Comment:
An empty string here is strange. Looking through the code, I'm not entirely sure why this is necessary or what it indicates. Could you explain?

Reply:
@fdmirza

Reply:
This was required for doing validation with rosetta-cli. The operations that are created through rosetta-cli for check:construction has empty status. So, the validation fails if we do not handle empty status. That's why we had to introduce this.

Reply:
https://github.com/coinbase/rosetta-cli/issues/180

Reply:
The problem is that this empty status field is introduced in two areas that don't make sense and can cause confusion. 
1. Adding an empty status here results in it surfacing to the non-rosetta API docs and clients.
2. In code below, an empty status is added as a database value even though such a value should never end up in the database. 

It seems like there would be a more appropriate way to handle this situation?

Reply:
@zone117x I understand the problem. It is an intentional behaviour in rosetta-cli for construction. As long as the transaction is not on-chain, its status need to be empty. So with the current implementation of rosetta-cli validation its required for us to have empty status for running check:construction. Its kind of a blocker for us. If you can suggest a better way to deal with this situation that would be great.